src/content/developers/tutorials/integrating-your-smart-contract-with-the-frontend/index.md

```diff
@@ -0,0 +1,920 @@
+
+When you open this cloned `hello-world-part-four` repository, you'll notice that it contains two folders: `starter-files` and `completed`.
+
+* `starter-files` contains the starter files \(essentially the React UI\) for this project. In this tutorial, **we will be working in this directory**, as you learn how to bring this UI to life by connecting it to your Ethereum wallet and the Hello World smart contract that you published on Etherscan in [Part 3](https://docs.alchemy.com/alchemy/tutorials/hello-world-smart-contract/submitting-your-smart-contract-to-etherscan).
+* `completed` contains the entire completed tutorial and is there for you as a **reference** **if you get stuck.**
+
+Next, open your copy of `starter-files` to your favorite code editor \(at Alchemy, we we're big fans of [VSCode](https://code.visualstudio.com/download)\), and then navigate into your `src` folder.
+
+All of the code we'll write will live under the `src` folder. We'll be editing the `HelloWorld.js` component and the `util/interact.js` javascript files to give our project Web3 functionality.
+
+## Step 2: Check out the starter files {#step-2-check-out-the-starter-files}
+
+Before we start coding, it's super important that we figure out what's already provided for us in the starter files.
+
+### Get your react project running
+
+Let's start by running the React project in our browser. The beauty of React is that once we have our project running in our browser, any changes we save will be updated live in our browser.
+
+To get the project running, navigate to the root directory of the `starter-files` folder, and the run `npm install` in your terminal to install the dependencies of the project:
+
+```bash
+cd starter-files
+npm install
+```
+
+Once those have finished installing, run `npm start` in your terminal:
+
+```bash
+npm start
+```
+
+Doing so should open [http://localhost:3000/](http://localhost:3000/) in your browser, where you'll see the frontend for our project. It should consist of one field \(a place to update the message stored in your smart contract\), a "Connect Wallet" button, and an "Update" button.
+
+If you try clicking "Connect Wallet" or "Update" buttons, you'll notice that they don't work—that's because we still need to program their functionality! :\)
+
+### The `HelloWorld.js` component
+
+### **NOTE:** Make sure you're in the `starter-files` folder and not the `completed` folder!
+
+Let's go back into the `src` folder in our editor and open the `HelloWorld.js` file. It's super important that we understand everything in this file, as it is the primary React component we will be working on.
+
+At the top of this file, you'll notice we have several import statements that are necessary to get our project running, including the React library, useEffect and useState hooks, some items from the `./util/interact.js` \(we'll describe them in more details soon!\), and the Alchemy logo. 🧙‍♂️
```

Comment:
```suggestion
At the top of this file, you'll notice we have several import statements that are necessary to get our project running, including the React library, useEffect and useState hooks, some items from the `./util/interact.js` (we'll describe them in more details soon!), and the Alchemy logo <Emoji text="🧙‍♂️" size={1} />
```

Reply:
Please ensure any emojis are used as described in [the style guide](https://ethereum.org/en/contributing/style-guide/#images)